msprime/simulations.py

```diff
@@ -98,6 +99,43 @@ def almost_equal(a, b, rel_tol=1e-9, abs_tol=0.0):
     return abs(a - b) <= max(rel_tol * max(abs(a), abs(b)), abs_tol)
 
 
+def model_factory(model, reference_size=1):
+    """
+    Returns a simulation model corresponding to the specified model and
+    reference size.
+    - If model is None, the default simulation model is returned with the
+      specified reference size.
+    - If model is a string, return the corresponding model instance
+      with the specified reference size
+    - If model is an instance of SimulationModel, return a copy of it.
+      In this case, if the model's reference_size is None, set it to the
+      parameter reference size.
+    - Otherwise return a type error.
+    """
+    model_map = {
+        "hudson": StandardCoalescent(reference_size),
+        "smc": SmcApproxCoalescent(reference_size),
+        "smc_prime": SmcPrimeApproxCoalescent(reference_size),
+        "dtwf": DiscreteTimeWrightFisher(reference_size)
```

Comment:
want to add the simple sweep case here now?

Reply:
It's not clear to me how we'd do this with the single sweep as it's parametric (we need s, etc). Happy to discuss options though.